src/core/solver.py

```
@@ -272,69 +268,186 @@ def add_scoring_objectives(self, strategy: ScoringStrategy) -> None:
+        """Guides the solver to decide on high-impact participants first."""
+        impacts = []
+        for i, p in enumerate(self.participants):
+            total_abs_score = sum(abs(s) for s in p.scores.values())
+            impacts.append((total_abs_score, i))
+
+        sorted_indices = [idx for _, idx in sorted(impacts, reverse=True)]
+
+        branching_vars = []
+        for p_idx in sorted_indices:
+            for g_idx in range(self.num_groups):
+                branching_vars.append(self.x[(p_idx, g_idx)])
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)